src/main/java/xyz/holocons/mc/holoitemsrevamp/enchantment/Splinter.java

```diff
@@ -0,0 +1,290 @@
+     */
+    private void tryCleanupSplinterData(Player player, Block newTrunk){
+        final var activeSplinter = splinters.get(player);
+        // if(!shouldCleanup)
+        if(activeSplinter != null && activeSplinter.scheduledSplinters != 0){
+            // there is still a splinter going, so don't cleanup until that one's done
+            return;
+        }
+
+        splinters.remove(player);
+
+        if(!isTrunkBlock(newTrunk)){
+            return;
+        }
+
+        ActiveSplinter newData = new ActiveSplinter();
+        newData.origin = newTrunk;
+        newData.remainingCharges = 32; // In the future we can make this based on enchLevel
+        newData.scheduledSplinters = 0;
+        splinters.put(player, newData);
+    }
+
+    private void handleTreeTrunkBroken(Player player, Block trunkBlock){
+        // TODO
+    }
+
+    private void handleTreeBranchBroken(Player player, Block branchBlock){
+        // TODO
+    }
+
+    private void handleShroomStemBroken(Player player, Block stemBlock){
+        // aka the shroom stem
+        // TODO
+    }
+
+    private void handleShroomBlockBroken(Player player, Block branchBlock){
+        // TODO
+    }
+
+    private void scheduleSplinterAbility(Player player, Block block) {
+        var data = splinters.get(player);
+        if(data.remainingCharges <= 0){
+            // No power left to schedule anything.
+            return;
+        }
+        int splinterDelay = data.scheduledSplinters;
+        // Update and save data
+        data.remainingCharges -= 1;
+        data.scheduledSplinters += 1;
+        splinters.put(player, data);
+
+        // Schedule a splinter
+        new BukkitRunnable(){
+            @Override
+            public void run() {
+                // Break the block
+                // TODO: Check if the player is holding a Splinter axe first.
+                //  I got the itemStack to check for splinter, but idk how to check for splinter.
+                var activeItem = player.getActiveItem();
+                player.breakBlock(block);
+
+                // ... then update the data.
+                var updatedData = splinters.get(player);
+                updatedData.scheduledSplinters -= 1;
+                splinters.put(player, updatedData);
+            }
+        }.runTaskLater(plugin, splinterDelay);
+    }
+
+    /**
+     * Gets the potential next-branch blocks from this starting branch block. The trunk block is used since branches
+     * don't "curl in" on theirself, they continue to go farther from the trunk. If the branch is actually part
+     * of the trunk, all 8 adjacent blocks plus the 8 additional branches are returned.
+     */
+    private List<Block> getPossibleBranches(Block trunk, Block branch){
+        // Since the main operations afaik will be append and iterator, this is slightly better than arraylist.
+        // Not by much.
+        List<Block> ret = new LinkedList<>();
```

Comment:
It's the opposite. If you're only appending and iterating, arraylist is better. LinkedList is better when you will remove elements from the middle of the list.